Automattic/wp-calypso
# JP Settings: Remove legacy regeneratePostByEmail() action and state

WIP, don't merge yet. Contains #23774, rebase after merging that.

### Testing Instructions

* Go to `calypso.localhost:3000/settings/writing/<yourJPsite>`
* Scroll down to the 'Publishing Tools' section
* Make sure that 'Publish posts by sending an email' is enabled
* Verify that the 'Email Address' input field contains a random-looking email address
* Click 'Regenerate'
* Verify that the 'Email Address' input field contains is updated to a _different_ random-looking email address
